include/swift/Reflection/TypeRefBuilder.h

```diff
@@ -725,8 +775,37 @@ class TypeRefBuilder {
       }),
       OpaquePointerReader([&reader](remote::RemoteAddress address, unsigned size) -> llvm::Optional<remote::RemoteAbsolutePointer> {
         return reader.Reader->readPointer(address, size);
+      }),
+      OpaqueIntVariableReader(
```

Comment:
This is the part I'm least happy with about this approach.  TypeRefBuilder has this pile of closures ostensibly to avoid making TypeRefBuilder itself a template, but it's getting increasingly unwieldy.  I'll take a stab at refactoring this a bit to make it cleaner, but any suggestions would be appreciated.  And of course, this "Int Variable Reader" compromises on the 32-bit/64-bit problem by requiring the caller to provide an explicit size, doing a 4-byte or 8-byte read as appropriate and then always returning the result in a `uint64_t`.